lib/ansible/modules/files/xml.py

```diff
@@ -0,0 +1,755 @@
+                # 'attrname' attribute for the key
+                parent.attrib.pop(result.attrname)
+            elif is_node(tree, xpath, namespaces):
+                # Delete an element
+                result.getparent().remove(result)
+            else:
+                raise Exception("Impossible error")
+    except Exception as e:
+        module.fail_json(msg="Couldn't delete xpath target: %s (%s)" % (xpath, e))
+    else:
+        finish(module, tree, xpath, namespaces, changed=True)
+
+
+def replace_children_of(children, match):
+    for element in match.getchildren():
+        match.remove(element)
+    match.extend(children)
+
+
+def set_target_children_inner(module, tree, xpath, namespaces, children, in_type):
+    matches = tree.xpath(xpath, namespaces=namespaces)
+
+    # Create a list of our new children
+    children = children_to_nodes(module, children, in_type)
+    children_as_string = [etree.tostring(c) for c in children]
+
+    changed = False
+
+    # xpaths always return matches as a list, so....
+    for match in matches:
+        # Check if elements differ
+        if len(match.getchildren()) == len(children):
+            for idx, element in enumerate(match.getchildren()):
+                if etree.tostring(element) != children_as_string[idx]:
+                    replace_children_of(children, match)
+                    changed = True
+                    break
+        else:
+            replace_children_of(children, match)
+            changed = True
+
+    return changed
+
+
+def set_target_children(module, tree, xpath, namespaces, children, in_type):
+    changed = set_target_children_inner(module, tree, xpath, namespaces, children, in_type)
+    # Write it out
+    finish(module, tree, xpath, namespaces, changed=changed)
+
+
+def add_target_children(module, tree, xpath, namespaces, children, in_type):
+    if is_node(tree, xpath, namespaces):
+        new_kids = children_to_nodes(module, children, in_type)
+        for node in tree.xpath(xpath, namespaces=namespaces):
+            node.extend(new_kids)
+        finish(module, tree, xpath, namespaces, changed=True)
+    else:
+        finish(module, tree, xpath, namespaces)
+
+_ident = "[a-zA-Z-][a-zA-Z0-9_\-\.]*"
+_nsIdent = _ident + "|" + _ident + ":" + _ident
+# Note: we can't reasonably support the 'if you need to put both ' and " in a string, concatenate
+# strings wrapped by the other delimiter' XPath trick, especially as simple XPath.
+_xpstr = "('(?:.*)'|\"(?:.*)\")"
+
+_re_splitSimpleLast = re.compile("^(.*)/(" + _nsIdent + ")$")
+_re_splitSimpleLastEqValue = re.compile("^(.*)/(" + _nsIdent + ")/text\\(\\)=" + _xpstr + "$")
+_re_splitSimpleAttrLast = re.compile("^(.*)/(@(?:" + _nsIdent + "))$")
+_re_splitSimpleAttrLastEqValue = re.compile("^(.*)/(@(?:" + _nsIdent + "))=" + _xpstr + "$")
+
+_re_splitSubLast = re.compile("^(.*)/(" + _nsIdent + ")\\[(.*)\\]$")
+
+_re_splitOnlyEqValue = re.compile("^(.*)/text\\(\\)=" + _xpstr + "$")
```

Comment:
I see... it's putting them near split_xpath_last which is where they're used.  Maybe move both the variables and the split_xpath_last function up to below the imports.